src/mlang/backend_ir/bir.mli

```diff
@@ -57,10 +57,17 @@ type mpp_function = { mppf_stmts : stmt list; mppf_is_verif : bool }
 
 module FunctionMap : Map.S with type key = function_name
 
+type program_context = {
```

Comment:
I'm ok with the change but you should put the rationale about why we need this context as a documentation comment here (you can start from your PR description which is quite informative).

Reply:
Documentation comment proposed in commit [b8e0f56](https://github.com/MLanguage/mlang/pull/184/commits/b8e0f56a440017d640a2a4646980fb974bbaa939).